hyperspy/axes.py

```diff
@@ -1071,17 +1075,23 @@ def convert_to_functional_data_axis(self, expression, units=None, name=None, **k
             d["name"] = name
         d.update(kwargs)
         this_kwargs = self.get_axis_dictionary()
+        self.remove_trait('scale')
+        self.remove_trait('offset')
         self.__class__ = FunctionalDataAxis
         d["_type"] = 'FunctionalDataAxis'
         self.__init__(expression=expression, x=UniformDataAxis(**this_kwargs), **d)
         self.axes_manager = axes_manager
 
     def convert_to_non_uniform_axis(self):
         d = super().get_axis_dictionary()
+        super()._init_trait_listeners()
```

Comment:
Why is this necessary?

Reply:
Good point, must have remained from one of my tries to get the traits removal working.

Reply:
(Now removed)